User can see list of posts

sealfury opened the pull request:
[CLIENT: User can see a list of all posts in a feed  ](https://www.pivotaltracker.com/story/show/176613149)
### User Story
```
As a User,            
In order to see what music people are sharing,         
I want to see a feed of posts. 
```
## Changes proposed in this pull request:
* adding `HomeScreen` component with an index of post
* adding a `PostService` module to perform CRUD actions
* adding `PostIndex` component which imports track name, artist, image, preview URL and description from backend 
* adding `TrackPlayer` component, making use of `expo-av` for playback functionality
* creating `cypress` test to test for API response
* creating `fixture` that mock API response
* creating `jest` component test for `HomeScreen`
* adding main header and App title 
* configuring navigation functionality
* configuring entryPoint to App across platforms 
* adding `StylesSheet` component
## Packages/Gems added
* `@react-navigation/native`
* `@react-navigation/stack`
* `react-native-screens`
* `react-redux` and `redux`
* `@react-native-community/masked-view`
* `expo-av@8.7.0`
* `vector-icons`
* `axios`
* `jest-expo`
* `@testing-library/react-native`
* `@testing-library/jest-native`
* `cypress`
## Screenshots (Client)
![Screenshot 2021-01-27 at 20 30 26](https://user-images.githubusercontent.com/30779084/106043746-233b6280-60df-11eb-8bb1-615e744f204b.png)

ViktorHek committed:
creating a fixture with post from backend. writing first test. creating postIndex component

ViktorHek committed:
adding react redux. configuring redux bu creating store, initialState and rootReducer

ViktorHek committed:
adding @react-navigation. creating homeScreen component and importing it in app.jsx. adding TuneShare as appTitle in initialState

sealfury committed:
removing imports of components not in use
fixing a typo

tochman commented on src/components/postIndex.jsx:
```diff
@@ -0,0 +1,14 @@
+  )
+}
+
+export default postIndex
```
So, the convention for React components is `PascalCase`

sealfury replied:
Of course, we'll fix this

sealfury committed:
adds redux-mock-store dev dependency
configures jest initialise action
creates fixture for mock of static data for use in component tests
creates test for HomeScreen component

sealfury committed:
finalizes component test for HomeScreen

sealfury committed:
reinstates test for App component using jest redux mocks

sealfury committed:
adds axios dependency
creates module for index action for user posts
updates root reducer for post index case
changes formatting to exclude semicolon

sealfury committed:
completes mapping function of posts data
adds components to render posts data as cards in flatlist
adds react-native=elements package
adds testID and data-cy props to components

sealfury committed:
corrects naming standard in cypress fixture
corrects stubbed url in cypress test
removes keyExtractor from FlatList

sealfury committed:
changes naming standard in cypress fixture

tochman commented on specs/App.spec.js:
```diff
@@ -1,11 +1,21 @@
+      </Provider>
+    )
+    expect(screen.getByText('TuneShare')).toBeTruthy()
+    expect(screen.getByText('TuneShare').props.style.fontSize).toEqual(17)
```
This was an example in my demo commit. I see no reason to actually use this assertion.

ViktorHek committed:
downgrading react-native-gesture-handler to 1.8.0 and react-native-screens to 2.15.2. importing jest fixture in cypress test. adding testID to homescreen

ViktorHek committed:
modifying cupress test to look for content in body. adding statickdata in homescreen spec and modifying contains_post_data toBe 3

ViktorHek committed:
modify labels on posts to match backend. moved the fixture. created stylse component.

ViktorHek committed:
good styling

sealfury committed:
adds expo-av package
creates functionality to play song preview for each track in index

ViktorHek committed:
creating trackPlayer component. adding stopPlayback button

sealfury committed:
finalizes functionality for start and stop playback of all songs in index
adds ternary rendering of play button based on playback status
moves playback button above post description field

ViktorHek committed:
replasing play button with plat icon. plasing play icon in post image. adding styling

ViktorHek committed:
reinstalled yarnlock

ViktorHek committed:
minor styling

sealfury committed:
adding styling to header container

sealfury committed:
testing expo vector icons

sealfury committed:
configures expo packages and reinstalls vector icons
finalises preliminary styling
exchanges anticons for ionicons
removes resolver module exports in metro config
formats code

sealfury committed:
removes commented code from jest test

KyunginNa commented on metro.config.js:
```diff
@@ -1,5 +1,3 @@
-  resolver: {
-    sourceExts: ['jsx', 'js']
-  }
+
```
Why is this file empty..?

sealfury replied:
the metro configuration was causing conflicts with the vector icons in the simulator render, so we had to remove them, but for safety's sake we left the file as there could be a little node module somewhere that depends on its existence, even if it's empty

KyunginNa commented on specs/HomeScreen.spec.js:
```diff
@@ -0,0 +1,65 @@
+import HomeScreen from '../src/components/HomeScreen'
+import initialState from '../src/state/store/initialState'
+
+const posts = [
```
You couldn't import the fixture file here?☺️

sealfury replied:
we tried this at first, but since the test relies on mocking quite a bit we ended up in a sort of jest-props-hell bottlenecked the feature, so we've left it as is- this will also help as a framework for future tests to help us follow jest's process.

KyunginNa commented on src/components/TrackPlayer.jsx:
```diff
@@ -0,0 +1,32 @@
+  }
+  async function stopPlayback() {
+    await playback.stopAsync()
+    setPlayback(undefined)
```
What is this line doing? (Curious!)

sealfury replied:
`playback` is actually an object in this case, and can't participate in the `stop`/`play`/`loadAsync()` functions unless it is.  So it can't be `null` or a boolean for the playback functionality to work.

KyunginNa reviewed (commented):
Left some questions!:)

sealfury committed:
Rename postIndex.jsx to PostIndex.jsx
